src/allmydata/scripts/tahoe_invite.py

```diff
@@ -0,0 +1,101 @@
+import json
+from os.path import join
+from ConfigParser import NoSectionError
+
+from twisted.python import usage
+from twisted.internet import defer, reactor
+
+from wormhole import wormhole
+
+from allmydata.util import configutil
+
+
+INVITE_SEPARATOR = "+"
+
+class InviteOptions(usage.Options):
+    synopsis = "[options] <nickname>"
+    description = "Create a client-only Tahoe-LAFS node (no storage server)."
+
+    optParameters = [
+        ("needed", None, None, "How many shares are needed to reconstruct files from this node"),
+        ("happy", None, None, "Distinct storage servers new node will upload shares to"),
+        ("total", None, None, "Total number of shares new node will upload"),
+    ]
+
+    def parseArgs(self, *args):
+        if len(args) != 1:
+            raise usage.UsageError(
+                "Provide a single argument: the new node's nickname"
+            )
+        self['nick'] = args[0].strip()
+
+
+@defer.inlineCallbacks
+def invite(options):
+    basedir = unicode(options.parent['node-directory'])
```

Comment:
`options.parent['node-directory'].decode('ascii')`?  `.decode('utf-8')`?  The former is probably almost exactly the same as `unicode(...)` but it makes the ASCII assumption strongly (Grr PyGtk!) and explicit.